compiler/rustc_passes/src/dead.rs

```diff
@@ -637,8 +636,8 @@ impl<'tcx> Visitor<'tcx> for MarkSymbolVisitor<'tcx> {
                 let res = self.typeck_results().qpath_res(path, pat.hir_id);
                 self.handle_field_pattern_match(pat, res, fields);
             }
-            PatKind::Path(ref qpath) => {
-                let res = self.typeck_results().qpath_res(qpath, pat.hir_id);
+            PatKind::Expr(PatExpr { kind: PatExprKind::Path(ref qpath), hir_id, .. }) => {
+                let res = self.typeck_results().qpath_res(qpath, *hir_id);
                 self.handle_res(res);
             }
```

Comment:
shouldn't this be in `visit_pat_expr` as this will not visit paths in range patterns

Reply:
I guess this would be pre-existing

Reply:
good catch, this was regressed in https://github.com/rust-lang/rust/pull/134228 (before that it was handled in `ExprKind::Path`

Reply:
Actually why is this code so complicated and fragile? Why aren't we just implementing `visit_qpath`?